src/main/java/nextstep/courses/service/SessionService.java

```diff
@@ -21,13 +21,35 @@ public class SessionService {
     @Resource(name = "sessionUsersRepository")
     private SessionUsersRepository sessionUsersRepository;
 
+    @Resource(name = "tutorRepository")
+    private TutorRepository tutorRepository;
+
+    @Resource(name = "userRepository")
+    private UserRepository userRepository;
+
     @Transactional
     public void register(long sessionId, NsUser user, Payment payment) {
         SessionUsers sessionUsers = sessionUsersRepository.findBy(sessionId);
-        Session session = sessionRepository.findBy(sessionId, sessionUsers)
+        Session session = sessionRepository.findBy(sessionId)
                 .orElseThrow(() -> new SessionException("강의 정보를 찾을 수 없습니다."));
         session.register(user, payment);
         sessionUsersRepository.addUserFor(sessionId, user.getId());
-        sessionRepository.updateCountBy(session.userCount(), sessionId);
     }
+
+    @Transactional
+    public void approve(long sessionId, NsTutor tutor, NsUser user) {
+        Session session = sessionRepository.findBy(sessionId)
+                .orElseThrow(() -> new SessionException("강의 정보를 찾을 수 없습니다."));
+        SessionUserStatus approve = tutor.approve(session, user);
+        sessionUsersRepository.updateSessionUserStatus(sessionId, user.getId(), approve);
+    }
```

Comment:
(참고만 해주세요)
승인 기능은 `수강신청한 사람`의 상태를 변경하는 기능이니
수강신청한 사람들을 알고 있는 객체에게 메시지를 보내서 승인된 `수강신청한 사람` 객체를 반환 받는 로직도 고려해 보시면 좋을 것 같아요

```suggestion
    @Transactional
    public void approve(long sessionId, NsTutor tutor, NsUser user) {
        Session session = sessionRepository.findBy(sessionId)
            .orElseThrow(() -> new SessionException("강의 정보를 찾을 수 없습니다."));

        SessionUser sessionUser = session.approve(tutor, user);

        sessionUsersRepository.save(sessionUser);
    }
```

Reply:
해당 로직으로 변경하려고 하니 관련되서 고칠께 굉장히 많아졌네요!! 그래도 이 방법이 좋은것 같습니다!! ㅎㅎ 감사합니다!